### PR TITLE
fix: removing protocolIDs from global namespace in p2p and fraud

### LINKED ISSUE #1189: all: don't populate protocol ids with chain_id in global space
Instead, each protocol like HeaderEx should have an option or config to receive and prepend a chain ID in the constructor and not during global initialization.

### PR DESCRIPTION
**Pull request by distractedm1nd**
Based on #1168 
Closes #1189

### CODE REVIEW

File: fraud/service_test.go [hunk 1]
```diff
@@ -263,5 +265,6 @@ func createServiceWithHost(
 		store.GetByHeight,
 		sync.MutexWrap(datastore.NewMapDatastore()),
 		enabledSyncer,
+		"private",
```
Comment (renaynay) [inline fraud/service_test.go:268]: `network.Private` everywhere pls

File: header/p2p/exchange_test.go [hunk 2]
```diff
@@ -99,15 +99,15 @@ func TestExchange_RequestByHash(t *testing.T) {
...
+	serv := NewExchangeServer(host, store, "private")
 	err = serv.Start(ctx)
 	require.NoError(t, err)
 	t.Cleanup(func() {
 		serv.Stop(context.Background()) //nolint:errcheck
 	})
 
 	// start a new stream via Peer to see if Host can handle inbound requests
-	stream, err := peer.NewStream(context.Background(), libhost.InfoFromHost(host).ID, exchangeProtocolID)
+	stream, err := peer.NewStream(context.Background(), libhost.InfoFromHost(host).ID, protocolID("private"))
```
Comment (renaynay) [inline header/p2p/exchange_test.go:110]: you can extract `protocolID("private")` into var

File: nodebuilder/fraud/fraud.go [hunk 3]
```diff
@@ -40,8 +43,9 @@ func newFraudService(
 	host host.Host,
 	hstore header.Store,
 	ds datastore.Batching,
-	isEnabled bool) (Module, error) {
-	pservice := fraud.NewProofService(sub, host, hstore.GetByHeight, ds, isEnabled)
+	isEnabled bool,
+	protocolSuffix string) (Module, error) {
```
Comment (renaynay) [inline nodebuilder/fraud/fraud.go:47]:
```suggestion
	protocolSuffix string,
	) (Module, error) {
```
format pls

File: fraud/service_test.go (hunk 1 above)
Comment (distractedm1nd) [reply fraud/service_test.go:268]: done

File: header/p2p/exchange_test.go (hunk 2 above)
Comment (distractedm1nd) [reply header/p2p/exchange_test.go:110]: done

File: nodebuilder/fraud/fraud.go (hunk 3 above)
Comment (distractedm1nd) [reply nodebuilder/fraud/fraud.go:47]: done

File: fraud/service_test.go (hunk 1 above)
Comment (distractedm1nd) [reply fraud/service_test.go:268]: reverted from @Wondertan 's sync review

**Review by renaynay — APPROVED** [review]

**Pull request merged**